paasta_tools/mesos_maintenance.py

```diff
@@ -0,0 +1,475 @@
+    try:
+        up_output = client_fn(method="POST", endpoint="/machine/up", data=json.dumps(payload)).text
+    except HTTPError as e:
+        e.msg = "Error performing maintenance up. Got error: %s" % e.msg
+        raise
+    return up_output
+
+
+def status():
+    """Get the Mesos maintenance status. This contains hostname/ip mappings for hosts that are either marked as being
+    down for maintenance or draining.
+    :returns: None
+    """
+    try:
+        status = get_maintenance_status()
+    except HTTPError as e:
+        e.msg = "Error performing maintenance status. Got error: %s" % e.msg
+        raise
+    return status.text
+
+
+def is_host_drained(hostname):
+    """Checks if a host has drained successfully by confirming it is
+    draining and currently running 0 tasks
+    :param hostname: hostname to check
+    :returns: True or False
+    """
+    mesos_state = get_mesos_state_summary_from_leader()
+    task_counts = get_mesos_task_count_by_slave(mesos_state)
+    if hostname in task_counts:
+        slave_task_count = task_counts[hostname].count
+    else:
+        slave_task_count = 0
+    return is_host_draining(hostname=hostname) and slave_task_count == 0
+
+
+def is_host_past_maintenance_start(hostname):
+    """Checks if a host has reached the start of its maintenance window
+    :param hostname: hostname to check
+    :returns: True or False
+    """
+    return hostname in get_hosts_past_maintenance_start()
+
+
+def is_host_past_maintenance_end(hostname):
+    """Checks if a host has reached the end of its maintenance window
+    :param hostname: hostname to check
+    :returns: True or False
+    """
+    return hostname in get_hosts_past_maintenance_end()
+
+
+def get_hosts_past_maintenance_start():
+    """Get a list of hosts that have reached the start of their maintenance window
+    :returns: List of hostnames
+    """
+    schedules = get_maintenance_schedule().json()
+    current_time = datetime_to_nanoseconds(now())
+    ret = []
+    if 'windows' in schedules:
+        for window in schedules['windows']:
+            if window['unavailability']['start']['nanoseconds'] < current_time:
+                ret += [host['hostname'] for host in window['machine_ids']]
+    log.debug("Hosts past maintenance start: {0}".format(ret))
+    return ret
+
+
+def get_hosts_past_maintenance_end():
+    """Get a list of hosts that have reached the end of their maintenance window
+    :returns: List of hostnames
+    """
+    schedules = get_maintenance_schedule().json()
+    current_time = datetime_to_nanoseconds(now())
+    ret = []
+    if 'windows' in schedules:
+        for window in schedules['windows']:
+            end = window['unavailability']['start']['nanoseconds'] + window['unavailability']['duration']['nanoseconds']
+            if end < current_time:
+                ret += [host['hostname'] for host in window['machine_ids']]
+    log.debug("Hosts past maintenance end: {0}".format(ret))
```

Comment:
You never return `ret`, so your test fails. Testing it locally, if you `return ret` after the `log.debug`, all of the unit tests pass.